Assets/AnkrSDK/Scripts/Ads/AnkrAds.cs

```diff
@@ -9,24 +11,95 @@ namespace AnkrSDK.Ads
 {
 	public static class AnkrAds
 	{
-		private const string PublicAddress = "public_address";
+		private const string DeviceIdFieldName = "device_id";
+		private const string AdTypeFieldName = "ad_type";
+		private const string AppIdFieldName = "app_id";
+		private const string PublicAddressFieldName = "public_address";
+		private const string LanguageFieldName = "language";
 
-		private const string RandomImageURLBase = "http://45.77.189.28:5001/ad";
+		private const string RandomImageURLBaseAD = "http://45.77.189.28:5001/ad";
+		private const string RandomImageURLBaseStart = "http://45.77.189.28:5001/start";
 
-		public static async UniTask<AdData> RequestAdData(string accountAddress, AdType adType)
+		public static async UniTask<AdData> Show(AdType adType)
 		{
-			var adRequestResult = await SendAdRequest(accountAddress);
-			return string.IsNullOrEmpty(adRequestResult.Error)
+			return await RequestAdData(SystemInfo.deviceUniqueIdentifier, adType);
+		}
+		
+		public static async UniTask<bool> Initialize()
+		{
+			return await RequestStartSessionData
+				("com.ankr.test", SystemInfo.deviceUniqueIdentifier, EthHandler.DefaultAccount,"EN"); //AppId : Application.identifier
+		}
+		
+		private static async UniTask<AdData> RequestAdData(string deviceId, AdType adType)
+		{
+			var adRequestResult = await SendAdRequest(deviceId, adType);
+			
+			switch (adRequestResult.Code)
```

Comment:
you have the same logic in both request ad and send start.
Move switch with debug.log into a method and add method-specific behavior after you call this method.